ocaml/tests/test_observer.ml

```diff
@@ -13,27 +13,114 @@
  *)
 open Tracing
 
-module D = Debug.Make (struct let name = "test_tracing" end)
+module D = Debug.Make (struct let name = "test_observer" end)
 
 open D
 
 let () = Printexc.record_backtrace true
 
-let _ =
-  create ~enabled:false ~name_label:"default" ~uuid:"123" ~tags:[]
-    ~endpoints:["bugtool"] ~processors:[] ~filters:[] ~service_name:"xapi" ;
-  main ()
+let test_trace_log_dir = "/tmp/var/log/dt/zipkinv2/json/"
+
+let () = Export.Destination.File.set_trace_log_dir test_trace_log_dir
+
+module Xapi_DB = struct
+  let assert_x_observers ~__context x =
```

Comment:
What is an `x_observer`?

Reply:
changed to assert_num_observers if that's more straighforward?